flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -414,9 +417,19 @@ public void enableUnalignedCheckpoints() {
 	}
 
 	/**
-	 * Returns whether checkpoints should be persisted externally.
+	 * Disables unaligned checkpoints.
 	 *
-	 * @return <code>true</code> if checkpoints should be externalized.
+	 * @see #enableUnalignedCheckpoints()
+	 */
+	@PublicEvolving
+	public void disableUnalignedCheckpoints() {
```

Comment:
Do we really need this method? Isn't `enableUnalignedCheckpoints(false);` enough? `enableUnalignedCheckpoints()` also is kind of redundant but probably more common.

For another `boolean` property we have just a single setter `setPreferCheckpointForRecovery`

Reply:
No we don't need it and I have no hard feelings for removing it. 

`disableUnalignedCheckpoints` would be relevant when UC becomes the default (see also `StreamExecutionEnvironment#disableOperatorChaining`).

Reply:
So what about keeping and using just `setUnalignedCheckpoints(boolean)`?

Reply:
Yes, I'd probably go this way. Should we mend that also on 1.11?

Reply:
Extracted FLINK-18403, which will be backported (while this PR should only go to master).